src/main/java/org/apache/maven/plugin/gpg/GpgVersionParser.java

```diff
@@ -37,12 +37,11 @@
  */
 public class GpgVersionParser
 {
-    private final GpgVersionConsumer consumer;
+    private final GpgVersion gpgVersion;
 
-    private GpgVersionParser( GpgVersionConsumer consumer )
+    private GpgVersionParser( GpgVersion gpgVersion )
```

Comment:
Conceptually this looks weird to me. A `GpgVersionParser` should get some kind of input and the result would be a `GpgVersion`. Maybe the consumer as input is overkill, another solution would be just a String.

Reply:
The class design is weird anyways: The GpgVersionParser creates/contains a GpgVersion. But, in fact, this is only the full line as extracted from the CommandLine executor. If you ask for the SemVer version, the GpgVersion class has first to do some parsing as well ...

Reply:
>  So why not encapsulate the full behaviour in GpgVersionParser (as the name suggests) and make GpgVersion a value type without any logic?

I fully agree on that

Reply:
Just keep in mind, that `GpgVersion` should represent the output of `gpg --version`.
And that is much more than the gpg version, like pubkey, Cipher, Hash, Compression.
Up until now I only needed to know the actual gpg version, so that's why only the first line is being parsed. But we might need to extend that class in the future.